content/articles/constructors_in_kotlin/index.md

```diff
@@ -0,0 +1,196 @@
+     val mystudent = myStudent("Alex",200) 
+    
+}
+```
+The output of the above code is:
+
+```
+Name = Alex 
+Id = 200
+Email = Alex@gmail
+```
+
+### Init blocks
+Init short for initializer. It is a block of code that is executed every time the class is instantiated.They are used when you need to add a block of code that cannot be added to constructor  which simplifies a lot for the programmer it is called immediately after the primary constructor. To declare one, use the `init` keyword:
+
+```kotlin
+init
+{ 
+    //code to be executed
+}
+```
+
+Here is an example of a primary constructor with an init block:
+
+```kotlin
+class person(name:string,id:Int){ 
+    val pName :String 
+    val pId:Int 
+    init{ 
+        pName= name.capitalize() 
+        pId=id println("Name = ${pName}") 
+        println("Id=${PId}") 
+        } 
+    } fun main(args:Array<Strings>){ 
+        val Person=person("Alex",200) 
+}
+```
+Explanation:
+When a person object is created, the parameters name and id accept the values "Alex" and "200."
+The properties name and id are not properties of the person class because they are not preceded by "val" or "var."
+
+
+When a person object is created, the initializer block is executed, which initializes pname and pId. 
+
+
+When ran this would output:
+
+```
+Name = Alex
+Id =200
+```
+
+### The difference between a primary constructor and a secondary constructor
+The main difference between a primary and secondary constructor is that the former resides in the class declaration header, whereas the latter is declared inside the Kotlin class body. Additionally, a secondary constructor may call the primary constructor.
+
+### Best practices for object-oriented programming
+The clean use of classes is required for effective Kotlin. We learned about constructors above but never discussed best practices with them. We can make the following changes to our code to make it more readable while using constructors:
+- Minimize visibility of properties, i.e., private, public, protected. It allows you to control where your class members can be accessed from.
+- Combine property declaration. This is achieved by taking a property and setting it to a constructor parameter.
+- Improve variable naming. This is achieved by using variable naming rules; your variable names should: 
+  - Be based on the variable's purpose so it's clear what the variable is for.
+  - Have no spaces between words. Ex. myClass.
+  - Be greater than a single character.
+  - Not use special prefixes or suffixes (e.g. name_, mName, s_name, and kName), except in the case of backing properties.
+
+### Conclusion
+In this article, we have covered various ways of creating constructors using Kotlin as our language, and when to use the init block. We also learned that the init block is called just after the primary constructor, and the secondary constructor is called using the `this` keyword. Finally, we went through best practices in object-oriented programming. 
```

Comment:
> We also learned that the init block is called just after the primary constructor, and the secondary constructor is called using the `this` keyword.

It's best to have your conclusion reserved for the bigger picture of what the guide was about and not the lower-level details. Maybe instead you could say something along the lines of:

 *we also learned when the init block is executed vs when the primary constructor is executed, along with how secondary constructors are invoked.*